### PR TITLE
defrag a base after closing with corresponding flags

### PR DESCRIPTION
**Pull request by s-mx**
There are EBLOB_AUTO_INDEX_SORT, EBLOB_AUTO_DATA_SORT flags and we want
to closed bases will be defragmentated in the near time. So the solution
is to enqueue such bases in the queue of next defragmentation and choose
first bases from the queue.

### CODE REVIEW

File: library/blob.h [hunk 1]
```diff
@@ -129,7 +129,8 @@ inline static int eblob_fdatasync(int fd)
 
 struct eblob_base_ctl {
 	struct eblob_backend	*back;
-	struct list_head	base_entry;
+	struct list_head	base_entry;  // entry of queue of all bases
+	struct list_head	queue_entry;  // entry of queue to sort after close.
```
Comment (sabramkin) [inline library/blob.h]: Is naming good enough? Maybe something like `closed_unsorted_base_entry` is more long, but could be more understandable (and no comments required).

File: library/blob.h [hunk 2]
```diff
@@ -415,6 +417,8 @@ struct eblob_backend {
 	struct list_head	bases;
 	int			max_index;
 
+	struct list_head	queue_to_sort;  // queue of closed bases to sort.
```
Comment (sabramkin) [inline library/blob.h]: The same way, I think that more preferable something like `closed_unsorted_bases` with no comment. Member name must be understandable.

**Review by sabramkin — COMMENTED** [review]
I haven't complete my review, but currently I don't have enough time. Think about comments, and then fix or dismiss.

File: library/blob.h (hunk 1 above)
Comment (s-mx) [reply library/blob.h]: done

File: library/blob.h (hunk 2 above)
Comment (s-mx) [reply library/blob.h]: done

**Pull request closed without merging**